snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedAssociationReferenceSets.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2024 B2i Healthcare, https://b2ihealthcare.com
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.b2international.snowowl.snomed.datastore.request;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.b2international.snowowl.core.ServiceProvider;
+import com.b2international.snowowl.core.domain.BranchContext;
+import com.b2international.snowowl.core.domain.IComponent;
+import com.b2international.snowowl.snomed.common.SnomedConstants.Concepts;
+import com.b2international.snowowl.snomed.datastore.index.entry.SnomedConceptDocument;
+
+/**
+ * Fetches and memoizes SNOMED CT Association Reference Set IDs.
+ * Works as a cache on top of multiple SNOMED CT specific requests. Requests can access this cache via {@link ServiceProvider#service(Class)} method.
+ * 
+ * @since 9.5
+ */
+public final class SnomedAssociationReferenceSets {
+
+	private final BranchContext context;
+	private Set<String> associationReferenceSets;
+
+	public SnomedAssociationReferenceSets(BranchContext context) {
+		this.context = checkNotNull(context, "context");
+	}
+	
+	public Set<String> get() {
+		if (associationReferenceSets == null) {
+			associationReferenceSets = SnomedRequests.prepareSearchConcept()
+					.all()
+					.filterByAncestor(Concepts.REFSET_ASSOCIATION_TYPE)
```

Comment:
It might not make that much of a difference, but a status filter and a reference set type filter could also be used here (which requires changing the concept search into a reference set search).

Reply:
We can add those, but keep in mind that inactive concepts don't have parents/ancestors so that's probably just good to have for clarity, regarding refset types that's a good point that returns only the true refsets with at least one member associated.

Reply:
Done.